apps/s_socket.c

```diff
@@ -123,12 +157,27 @@ int init_client(int *sock, const char *host, const char *port,
     }
 
     if (*sock == INVALID_SOCKET) {
+        if (bindaddr != NULL && !found) {
+            BIO_printf(bio_err, "Can't bind %saddress for %s%s%s\n",
+                       BIO_ADDRINFO_family(res) == AF_INET6 ? "IPv6 " :
+                       BIO_ADDRINFO_family(res) == AF_INET ? "IPv4 " :
+                       BIO_ADDRINFO_family(res) == AF_UNIX ? "unix " : "",
```

Comment:
Styling/readability nit/side note. [For reference, "side note" customarily means that no action is actually expected.] Who can with confidence[!] tell what happens in above three lines in say 5 seconds? Intuitively you get the idea, but reviewer has to be confident ;-) As for styling. Here it doesn't *really* matter, but calling same function three times in tight row is kind of bad style. Caching as `int family = BIO_ADDINFO_family(ret)` would be sign of better understanding program structure. Once again, it's a side note. It wouldn't be if it allowed to write the chain condition in one line, but it doesn't. Nor is it performance-critical path...

Reply:
> Here it doesn't really matter, but calling same function three times in tight row is kind of bad style. 

gcc:  I find your lack of faith... disturbing.

Yeah, ok, but micro-optimising non-critical path code...

Reply:
> gcc: I find your lack of faith... disturbing.

Compiler is not in position to optimize this. Well, whole program IPA can, but customarily this functions would be in shared library...

> Yeah, ok, but micro-optimising non-critical path code...

Come on! I did say that it doesn't *really* matter because it's *not* performance-critical code. And that doing something *is* optional...

Reply:
It's yucky.  That's a technical term :)  But I agree with Andy, it doesn't have to change.  (And if it did, a BIO_to_afstring or something might be the way to go)